File: backend/src/main/java/lan/dk/podcastserver/manager/worker/noop/PassThroughExtractor.java
package lan.dk.podcastserver.manager.worker.noop;

import lan.dk.podcastserver.entity.Item;
import lan.dk.podcastserver.manager.worker.Extractor;
import lan.dk.podcastserver.manager.downloader.DownloadingItem;
import lombok.extern.slf4j.Slf4j;
import org.springframework.context.annotation.Scope;
import org.springframework.stereotype.Component;

import static io.vavr.API.List;
import static java.lang.Integer.MAX_VALUE;

/**
 * Created by kevin on 03/12/2017
 */
@Slf4j
@Component
@Scope("prototype")
public class PassThroughExtractor implements Extractor {
    
    @Override
    public DownloadingItem extract(Item item) {
        return DownloadingItem.builder()
                .item(item)
                .urls(List(item.getUrl()))
                .build();
    }

    @Override
    public Integer compatibility(String url) {
        return MAX_VALUE-1;
    }

}

File: backend/src/main/kotlin/com/github/davinkevin/podcastserver/manager/worker/noop/PassThroughExtractor.kt
package com.github.davinkevin.podcastserver.manager.worker.noop

import com.github.davinkevin.podcastserver.manager.downloader.DownloadingInformation
import com.github.davinkevin.podcastserver.manager.downloader.DownloadingItem
import com.github.davinkevin.podcastserver.manager.worker.Extractor
import org.springframework.context.annotation.Scope
import org.springframework.stereotype.Component
import java.lang.Integer.MAX_VALUE
import java.net.URI
import kotlin.io.path.Path

/**
 * Created by kevin on 03/12/2017
 */
@Component
@Scope("prototype")
class PassThroughExtractor : Extractor {

    override fun extract(item: DownloadingItem): DownloadingInformation {
        val fileName = Path(item.url.path).fileName
        return DownloadingInformation(item, listOf(item.url), fileName, null)
    }


    override fun compatibility(url: URI): Int = MAX_VALUE - 1

}
